injector/network_disruption.go

```diff
@@ -421,40 +427,63 @@ func (i *networkDisruptionInjector) applyOperations() error {
-				return fmt.Errorf("can't create the fw filter: %w", err)
+			// create a fourth prio with only 2 bands to filter traffic with a specific mark
+			if err := i.config.TrafficController.AddPrio(interfaces, "3:2", "4:", 2, [16]uint32{}); err != nil {
+				return fmt.Errorf("can't create a new qdisc: %w", err)
 			}
 
-			// create fw eBPF filter to classify packets based on http method and/or path
-			if err := i.config.TrafficController.AddBPFFilter(interfaces, "2:0", "/usr/local/bin/bpf-network-tc-filter.bpf.o", "2:2"); err != nil {
-				return fmt.Errorf("can't create the fw filter: %w", err)
+			// create fw eBPF filters to classify packets based on http method
+			if err := i.config.TrafficController.AddBPFFilter(interfaces, "2:0", "/usr/local/bin/bpf-network-tc-filter.bpf.o", "2:2", "classifier_methods"); err != nil {
+				return fmt.Errorf("can't create the eBPF fw filter: %w", err)
 			}
 
-			// run the program responsible to configure the map of the eBPF tc filter
+			// create fw eBPF filters to classify packets based on http path
+			if err := i.config.TrafficController.AddBPFFilter(interfaces, "3:0", "/usr/local/bin/bpf-network-tc-filter.bpf.o", "3:2", "classifier_paths"); err != nil {
+				return fmt.Errorf("can't create the eBPF fw filter: %w", err)
+			}
+
+			// run the program responsible to configure the maps of the eBPF tc filters
 			bpfConfigExecutor := network.NewBPFTCFilterConfigExecutor(i.config.Log, i.config.Disruption.DryRun)
-			err = i.config.TrafficController.ConfigBPFFilter(bpfConfigExecutor, "-f", i.spec.HTTP.Path, "-m", strings.ToUpper(i.spec.HTTP.Method))
+			configBPFFilterArgs := []string{}
 
-			if err != nil {
+			for _, path := range i.spec.HTTP.Paths {
+				configBPFFilterArgs = append(configBPFFilterArgs, "--path", string(path))
+			}
+
+			for _, method := range i.spec.HTTP.Methods {
+				configBPFFilterArgs = append(configBPFFilterArgs, "--method", strings.ToUpper(method))
+			}
+
+			if err = i.config.TrafficController.ConfigBPFFilter(bpfConfigExecutor, configBPFFilterArgs...); err != nil {
 				return fmt.Errorf("could not update the configuration of the bpf-network-tc-filter filter: %w", err)
 			}
 
-			// parent 3:2 refers to the 2nd band of the 3nd prio qdisc
-			// handle starts from 4 because 1, 2 are used by the 3 prio qdiscs
-			parent = "3:2"
-			handle = uint32(4)
+			// create fw filter to classify packets based on their mark
+			if err := i.config.TrafficController.AddFwFilter(interfaces, "4:0", types.InjectorCgroupClassID, "4:2"); err != nil {
+				return fmt.Errorf("can't create the fw filter: %w", err)
+			}
+
+			// parent 4:2 refers to the 3nd band of the 4th prio qdisc
+			// handle starts from 5 because 1, 2 and 3 are used by the 4 prio qdiscs
+			parent = "4:2"
+			handle = uint32(5)
 		} else {
+			// create second prio with only 2 bands to filter traffic with a specific mark
+			if err := i.config.TrafficController.AddPrio(interfaces, "1:4", "2:", 2, [16]uint32{}); err != nil {
```

Comment:
Why do we need this new Prio for non-http filters?

Reply:
I see this entered in `126e17192ffc5293ddd17823b52e9de0f4894ca3`, seems unexpected?

Reply:
Before this commit this line was applied for both scenarios: 
**Without http filters**: this qdisc is used to check if the mark is applied
**With http filters**: this qdisc is used to check the http method.
So I duplicated this line and add different comments depending of the scenario.